src/components/profile/household/crds-marital-status/crds-marital-status.tsx

```diff
@@ -81,58 +99,159 @@ export class UserMaritalStatus {
+      value: event.target.options[event.target.selectedIndex].title
+    };
+    if (this.originalValue) this.modalCheck(this.value.maritalStatus);
     this.formValueChange.emit();
+    if (event.target.value == 2) this.handleConfirmYes();
   }
 
-  private async handleConfirmCancel() {
+  private handleConfirmCancel = () => {
     this.formReset();
-    this.value.id = this.defaultValueID;
+    this.value = {
+      maritalStatus: this.originalValue,
+      spouse: this.value.spouse
+    };
     this.showConfirmationModal = false;
-  }
+  };
 
-  private handleModalClose = () => {
+  private handleConfirmYes = () => {
+    this.isDirty = true;
+    this.inputElement.classList.add('touched');
+    this.originalValue = this.value.maritalStatus;
     this.showConfirmationModal = false;
+    this.maritalStatusUpdate.emit(this.value.maritalStatus);
   };
 
+  private modalCheck(selectedMaritalStatus) {
+    // :: Marital Status Values ::
+    // Single: 1
+    // Married: 2
+    // Divorced: 3
+    // Widowed: 4
+    // Separated: 6
+    const originalMaritalStatusId = this.originalValue.id;
+    const newMaritalStatusId = selectedMaritalStatus.id;
+
+    if (originalMaritalStatusId == 2) {
+      if (newMaritalStatusId == 1 || newMaritalStatusId == 3 || newMaritalStatusId == 6) {
+        this.modalTextContent = this.renderModalMessageRemoveSpouse();
+        this.showConfirmationModal = true;
+      }
+      if (newMaritalStatusId == 4) {
+        // this.modalTextContent = this.renderModalMessageConfirmationOnly();
+        // this.showConfirmationModal = true;
+      }
+    } else {
+      if (newMaritalStatusId == 2) {
+        this.maritalStatusOption = false;
+      }
+      if (newMaritalStatusId == 3 || newMaritalStatusId == 6 || newMaritalStatusId == 1 || newMaritalStatusId == 4) {
+        // this.modalTextContent = this.renderModalMessageConfirmationOnly();
+        // this.showConfirmationModal = true;
+      }
+    }
+  }
+
+  private renderModalMessageRemoveSpouse() {
+    return (
+      <span>
+        {this.contentBlockService.getContentBlockText('maritalStatusSaveConfirmation')}{' '}
+        <b>{this.value.maritalStatus && this.value.maritalStatus.value}</b>?{' '}
+        {this.contentBlockService.getContentBlockText('maritalStatusRemoveSpouseInfo')}
+      </span>
+    );
+  }
+
+  private renderModalMessageConfirmationOnly() {
+    return (
+      <span>
+        {this.contentBlockService.getContentBlockText('maritalStatusSaveConfirmation')}{' '}
+        <b>{this.value.maritalStatus && this.value.maritalStatus.value}</b>?
+      </span>
+    );
+  }
+
+  private setMaritalStatusOption() {
+    this.maritalStatusOption = !this.maritalStatusOption;
+    console.log('this.maritalStatusOption', this.maritalStatusOption);
+    if (this.maritalStatusOption) {
+      Utils.setCookie('maritalStatusOption', 'true', 365);
+    } else {
+      Utils.setCookie('maritalStatusOption', 'false', 365);
+    }
```

Comment:
I THINK this could be rewritten as 
```suggestion
    Utils.setCookie('maritalStatusOption', this.maritalStatusOption, 365);
```
LMK what you think

Reply:
done